cppapi/server/except.cpp

```diff
@@ -45,6 +45,19 @@ namespace Tango
 char Except::mess[256];
 omni_mutex Except::the_mutex;
 
+std::ostream &operator<<(std::ostream &ss, const Tango::DevFailed &devFailed)
+{
+    for (int i = 0, size = devFailed.errors.length(); i < size; ++i)
+    {
+        DevError error = devFailed.errors[i];
+        ss << "\tOrigin:" << error.origin << "\n";
+        ss << "\tSeverity:" << error.severity << "\n";
+        ss << "\tReason:" << error.reason << "\n";
+        ss << "\tDescription:" << error.desc << "\n";
```

Comment:
Do you intend to use this operator in future as a generic utility for printing DevError or is it just for exception handler in dserver's init_hook? In former case, would concurrent access be a problem? In this example we will get mixed lines:
```
thread 1: std::cout << DevFailed{ ... } << std::cout;
thread 2: std::cout << DevFailed{ ... } << std::cout;
```
Buffering parts of the output in `ostringstream` before writing may help (probably depends on whether underlying `write` call is synchronized at OS level).

Reply:
It is intended to use this operator on a stack trace local stringstream, as it is just a convenience function. I will mark it as **inline**.